examples/gpu/eemumu_AV/SubProcesses/P1_Sigma_sm_epem_mupmum/check.cc

```diff
@@ -41,6 +42,38 @@ int usage(char* argv0, int ret = 1) {
   return ret;
 }
 
+#ifdef __CUDACC__
+template<typename T = fptype>
+struct CudaDevDeleter {
+  void operator()(T* mem) {
+    checkCuda( cudaFree( mem ) );
+  }
+};
+template<typename T = fptype>
+std::unique_ptr<T, CudaDevDeleter<T>> devMakeUnique(std::size_t N) {
+  T* tmp = nullptr;
+  checkCuda( cudaMalloc( &tmp, N * sizeof(T) ) );
+  return std::unique_ptr<T, CudaDevDeleter<T>>( tmp );
+}
+template<typename T = fptype>
+struct CudaHstDeleter {
+  void operator()(T* mem) {
+    checkCuda( cudaFreeHost( mem ) );
+  }
+};
+template<typename T = fptype>
+std::unique_ptr<T[], CudaHstDeleter<T>> hstMakeUnique(std::size_t N) {
```

Comment:
I would still like to see hstMakeUnique and hstMakeUniquePinned, the former without cudaMallocHost, the latter with cudaMallocHost

Reply:
Hm maybe I take this back now. I see that you are using hstMakeUnique here while in line 234 below you use explicitly a unique pointer. Looks better like you have done it, sorry for the noise. We should just remember to call this only for stuff that becomes pinned memory in cuda. Not sure, maybe we will have a mixture of pinned and not pinned in the future? But fo rthe momet ok like you are doing it.